app/src/main/java/org/simple/clinic/bp/assignbppassport/BpPassportEffectHandler.kt

```diff
@@ -1,23 +1,46 @@
 package org.simple.clinic.bp.assignbppassport
 
 import com.spotify.mobius.rx2.RxMobius
+import com.squareup.inject.assisted.Assisted
 import com.squareup.inject.assisted.AssistedInject
+import dagger.Lazy
+import io.reactivex.Observable
 import io.reactivex.ObservableTransformer
+import org.simple.clinic.facility.Facility
 import org.simple.clinic.patient.PatientRepository
 import org.simple.clinic.util.scheduler.SchedulersProvider
 
 class BpPassportEffectHandler @AssistedInject constructor(
     private val schedulersProvider: SchedulersProvider,
-    private val patientRepository: PatientRepository
+    private val patientRepository: PatientRepository,
+    private val currentFacility: Lazy<Facility>,
+    @Assisted private val uiActions: BpPassportUiActions
 ) {
 
+  @AssistedInject.Factory
+  interface Factory {
+    fun create(uiActions: BpPassportUiActions): BpPassportEffectHandler
+  }
+
   fun build(): ObservableTransformer<BpPassportEffect, BpPassportEvent> {
     return RxMobius
         .subtypeEffectHandler<BpPassportEffect, BpPassportEvent>()
         .addTransformer(SaveNewOngoingPatientEntry::class.java, saveNewPatientEntry())
+        .addTransformer(OpenPatientEntryScreen::class.java, openPatientEntryScreen())
         .build()
   }
 
+  private fun openPatientEntryScreen(): ObservableTransformer<OpenPatientEntryScreen, BpPassportEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map { currentFacility.get() }
```

Comment:
Here we are doing 2 tasks, getting current facility and opening the patient entry screen. Let's separate them out into 2 separate effects, WDYT?

Reply:
Makes sense. Will update it to have two separate effects.